<launch>
    <!-- Define arguments with default values -->
    <arg name="image_topic" default="/camera/image_raw" />
    <arg name="camera_info_topic" default="/camera/camera_info" />
    <arg name="parameter_file" default="$(find orb_slam2_ros)/config/mono.yaml" />

    <arg name="load_map" default="false" />
    <arg name="map_file_name" default="$(find orb_slam2_ros)/resource/map.bin" />

    <arg name="publish_map" default="true" />
    <arg name="publish_rendered_image" default="true" />
    <arg name="publish_pose" default="true" />
    <arg name="publish_tf" default="false" />

    <node name="orb_slam2_mono" pkg="orb_slam2_ros" type="orb_slam2_ros_mono" output="screen">
        <!-- Use the argument for remapping -->
        <remap from="/camera/image_raw" to="$(arg image_topic)" />
        <remap from="/camera/camera_info" to="$(arg camera_info_topic)" />

        <!-- static parameters -->
        <param name="publish_map" type="bool" value="$(arg publish_map)" />
        <param name="publish_rendered_image" type="bool" value="$(arg publish_rendered_image)" />
        <param name="publish_pose" type="bool" value="$(arg publish_pose)" />
        <param name="publish_tf" type="bool" value="$(arg publish_tf)" />

        <param name="load_map" type="bool" value="$(arg load_map)" />
        <param name="map_file_name" type="string" value="$(arg map_file_name)" />

        <param name="vocabulary_file_name" type="string" value="$(find orb_slam2_ros)/orb_slam2_core/Vocabulary/ORBvoc.txt" />
        <param name="parameter_file" type="string" value="$(arg parameter_file)" />
    </node>
</launch>